src/systems/optical_tactile_plugin/OpticalTactilePlugin.cc

```diff
@@ -0,0 +1,702 @@
+    }
+  }
+
+  if (!_sdf->HasElement("visualize_forces"))
+  {
+    ignlog << "Missing parameter <visualize_forces>, "
+      << "setting to " << this->dataPtr->visualizeForces << std::endl;
+  }
+  else
+  {
+    this->dataPtr->visualizeForces = _sdf->Get<bool>("visualize_forces");
+  }
+
+  if (!_sdf->HasElement("extended_sensing"))
+  {
+    ignlog << "Missing parameter <extended_sensing>, "
+      << "setting to " << this->dataPtr->extendedSensing << std::endl;
+  }
+  else
+  {
+    if (_sdf->Get<double>("extended_sensing") < 0)
+    {
+      ignwarn << "Parameter <extended_sensing> must be positive, "
+        << "setting to " << this->dataPtr->extendedSensing << std::endl;
+    }
+    else
+    {
+      this->dataPtr->extendedSensing = _sdf->Get<double>("extended_sensing");
+    }
+  }
+
+  if (!_sdf->HasElement("visualize_sensor"))
+  {
+    ignlog << "Missing parameter <visualize_sensor>, "
+      << "setting to " << this->dataPtr->visualizeSensor << std::endl;
+  }
+  else
+  {
+    this->dataPtr->visualizeSensor = _sdf->Get<bool>("visualize_sensor");
+  }
+
+  if (!_sdf->HasElement("force_length"))
+  {
+    ignlog << "Missing parameter <force_length>, "
+      << "setting to " << this->dataPtr->forceLength << std::endl;
+  }
+  else
+  {
+    if (_sdf->Get<double>("force_length") < 0)
+    {
+      ignwarn << "Parameter <force_length> must be positive, "
+        << "setting to " << this->dataPtr->forceLength << std::endl;
+    }
+    else
+    {
+      this->dataPtr->forceLength = _sdf->Get<double>("force_length");
+    }
+  }
+
+  // Get the size of the sensor from the SDF
+  // If there's no <collision> specified inside <link>, a default one
+  // is set
+  if (_sdf->GetParent() != nullptr)
+  {
+    if (_sdf->GetParent()->GetElement("link") != nullptr)
+    {
+      if (_sdf->GetParent()->GetElement("link")->GetElement("collision")
+          != nullptr)
+      {
+        if (_sdf->GetParent()->GetElement("link")->GetElement("collision")->
+            GetElement("geometry") != nullptr)
+        {
+          if (_sdf->GetParent()->GetElement("link")->GetElement("collision")->
+              GetElement("geometry")->GetElement("box") != nullptr)
+          {
+            this->dataPtr->sensorSize =
+              _sdf->GetParent()->GetElement("link")->GetElement("collision")->
+              GetElement("geometry")->GetElement("box")->
+              Get<ignition::math::Vector3d>("size");
+          }
+        }
+      }
+    }
+  }
+}
+
+//////////////////////////////////////////////////
+void OpticalTactilePlugin::PreUpdate(const UpdateInfo &_info,
+  EntityComponentManager &_ecm)
+{
+  IGN_PROFILE("OpticalTactilePlugin::PreUpdate");
+
+  // Nothing left to do if paused
+  if (_info.paused)
+    return;
+
+  if (!this->dataPtr->initialized)
+  {
+    // We call Load here instead of Configure because we can't be guaranteed
+    // that all entities have been created when Configure is called
+    this->dataPtr->Load(_ecm);
```

Comment:
My suggestion was just for `Load`. `components::ContactSensorData` may not be available during `Configure` because the `Contact` system creates it in `PreUpdate`. Either way, I suggest checking if `csd` is null before doing anything with it because it's not guaranteed to exist.
https://github.com/ignitionrobotics/ign-gazebo/blob/b321a0044e4feab3e6a1406139691581b90ad892/src/systems/optical_tactile_plugin/OpticalTactilePlugin.cc#L326  This would be the case, for example, if the `Contact` system was not added to the SDF file.

Reply:
The added benefit of calling `Load` in `Configure` is you don't have to check `initErrorPrinted`. `Configure` is called only once.

Reply:
@mcres Friendly ping. Do you have time to address the comment above? Looks like that is the last comment (and the previous Mac compiler error on CI, which is rerunning now), unless @azeey has more to review.
Let us know if you want us to help with the branch. Ideally, we'd like to get this in before code freeze on Sep 23.

Reply:
Okay I see what's going on here. @azeey So I tried moving the `Load()` call into `Configure()`, and as @mcres mentioned, the `csd` pointer in `PreUpdate()` is then just always null. I left it running for a while, and it never went non-null.

The same approach is used in the `TouchPlugin`, to call `Load()` from `PreUpdate()` instead of `Configure()`. I guess it must be similar here?
https://github.com/ignitionrobotics/ign-gazebo/blob/main/src/systems/touch_plugin/TouchPlugin.cc#L368-L374

Reply:
Ah, the issue is that we're checking for `ContactSensorData` in `Load` to determine the Collision associated with a contact sensor. That component won't be available until the `Contact` system has ran once.
https://github.com/ignitionrobotics/ign-gazebo/blob/b17e904d5d7884aa592ce3e4f38019f03f866bb1/src/systems/optical_tactile_plugin/OpticalTactilePlugin.cc#L402-L403

So, yeah, you guys are correct. `Load` needs to be in `PreUpdate`. And since currently, world systems are ran before model systems, the `Contact` system is guaranteed to run before the `OpticalTactilePlugin`.

Reply:
Cool! Thanks @azeey for looking into it. Could you approve the PR? I think it won't let us merge if there are remaining "Requested changes" marks. I made a PR to the fork to fix the Mac warning.